tljh-voila-gallery/tljh_voila_gallery/systemd-units/tljh-voila-gallery-builder.service

```diff
@@ -11,6 +11,9 @@ PrivateTmp=yes
 PrivateDevices=yes
 ProtectKernelTunables=yes
 ProtectKernelModules=yes
+# Update with the new gallery
+ExecStartPre={python_interpreter_path} -m pip install -U {gallery_repo}
```

Comment:
This looks like the simplest / easiest way to execute one more command.
If we plan on adding more logic to this unit in the future we can consider moving it to a proper script.

Reply:
hmm, I think this needs to be a separate process, since you'd need to restart the hub process too when this happens.

*or*, we can split the repo into two, code and data. Thta's the right thing to do long term anyway... we can then unconditionally pull in the data

Reply:
what does data mean in that case?

Reply:
Data would be gallery.yaml

Reply:
Then we are back to the question of what happens when the structure of the yaml changes and requires the new code that goes with it.

Reply:
Ugh, yeah.

But now when the code changes, the files will change but the running processes will not... Immutable deployment problems.

I don't have a clear answer, but if this unblocks you I'm happy to merge as long as we open an issue talking about what is the way out.

Reply:
Exactly, this would require a restart.

Reply:
As a side note this change also does not cover the case when the hooks have been updated and should be run one more time through the `--plugin` mechanism of TLJH. For example when adding more custom actions in `post_install_hook`.

In that case it would probably be better to bootstrap a complete new instance and tear down the old one.